impress-org/give-activecampaign
# Refactor: ActiveCampaign has block requirements

Resolves [GIVE-446]

## Description
- Blocks/Fields like ActiveCampaign have requirements to work.  They rely on being connected to an external service - in this case using the ActiveCampaign api.  These are things we need to check before trying to load the block in the form builder and also the field in the donation form.

This PR uses the `activecampaign->credentials_test()` to determine whether to show the block or not. In the builder if it is not valid an invalid placeholder as well as a notice in the inspector will display. Both the placeholder & the notice have links that will direct users to the Settings page to complete setup for the block.

Additional, changes:
- Update on how window data is passed to the form builders edit function.
- Refactor in EnqueueFormBuilderScripts.php to obtain tags from a separate method.
- Updates to the block placeholder design for consistency with other mailer add-ons.
- Enqueued stylesheet for the formbuilder.

## Affects
VFB
ActiveCampaign block

## Visuals

https://github.com/impress-org/give-activecampaign/assets/75056371/24d36aae-a83e-4143-811a-cf39a861e8a4


## Testing Instructions
- Activate ActiveCampaign 
- Try adding the ActiveCampaign block without setting up the apikey or api url, it should display the invalid block placeholder & notice.
- If you have Enable globally set from the settings page the block is usually is added to any new form. If no apikey is available the block should not be added.
- Test that the links on the placeholder and the invalid block direct users to the correct settings page.
- Setup the ActiveCampaign settings by adding your key, the block should now be able to be added as normal for both existing and new forms.

## Pre-review Checklist






[GIVE-239]: https://stellarwp.atlassian.net/browse/GIVE-239?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[GIVE-446]: https://stellarwp.atlassian.net/browse/GIVE-446?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
